components/automate-deployment/pkg/client/restore.go

```diff
@@ -471,6 +478,7 @@ func (r *DeploymentRestore) Restore() error {
 	// Make the spec actually execute commands
 	spec := *r.metadata.Spec
 	backup.SetCommandExecutor(spec, command.NewExecExecutor())
+	ctx, cancel := context.WithCancel(ctx)
```

Comment:
Ultimately, I doubt it matters because the parent context of this is going to get canceled at the end of this request, but I think in the general case, not calling cancel on this context is a leak of sorts since the executor only cancels contexts on error unless I've missed something.

Reply:
I agree. I think we made a mistake passing the cancel function to the executor. If it wants to be able to cancel stuff, it should create its own context with a cancel func. I'm guessing this does leak though given that we were canceling before

Reply:
also, this isn't the server code, so whatever